src/gui/BandSplitter/BandSplitterChyron.cpp

```diff
@@ -0,0 +1,95 @@
+#include "BandSplitterChyron.h"
+
+namespace gui::band_splitter
+{
+BandSplitterChyron::BandSplitterChyron (chowdsp::PluginState& pluginState,
+                                        dsp::band_splitter::Params& bandSplitterParameters,
+                                        const chowdsp::HostContextProvider& hcp)
+    : state (pluginState),
+      bandSplitterParams (bandSplitterParameters),
+      hostContextProvider (hcp)
+{
+    updateValues();
+    callbacks +=
+        {
+            pluginState.addParameterListener (*bandSplitterParams.threeBandOnOff,
+                                              chowdsp::ParameterListenerThread::MessageThread,
+                                              [this]
+                                              {
+                                                  updateValues();
+                                              }),
+            pluginState.addParameterListener (*bandSplitterParams.fourBandOnOff,
+                                              chowdsp::ParameterListenerThread::MessageThread,
+                                              [this]
+                                              {
+                                                  updateValues();
+                                              }),
+        };
+}
+
+void BandSplitterChyron::updateValues()
+{
+    const auto reset = [this]
+    {
+        cutoffSliderLow.reset();
+        cutoffSliderMid.reset();
+        cutoffSliderHigh.reset();
+    };
+
+    reset();
```

Comment:
Do we need the lambda here?

Reply:
I think so? To ensure that the optionals are reset when updateValues is called from the callbacks? Let me know if there is a preferable approach.

Reply:
Right, I agree that the optionals need to be reset, but it seems like we're only invoking the lambda in this one place... so couldn't we just write the `reset()` calls directly instead of wrapping them in a lambda?

Reply:
yes, quite right!